compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -2223,6 +2223,21 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                                         Err(LitToConstError::TypeError) => todo!(),
                                     }
                                 }
+
+                                hir::ExprKind::Path(hir::QPath::Resolved(
+                                    _,
+                                    &hir::Path {
+                                        res: Res::Def(DefKind::ConstParam, def_id), ..
+                                    },
+                                )) => {
+                                    let ty = tcx.type_of(def_id).instantiate_identity();
```

Comment:
```suggestion
                                    let ty = tcx.type_of(def_id).no_bound_vars().expect("const parameter types cannot be generic");
```

Reply:
Ideally we would factor out the logic common between this here and `lower_const_param` 🤔, not blocking through

Reply:
Slowly wondering if we should just use entirely new hir nodes so we can just use AnonConst like other type system consts